### ISSUE TITLE
OpenSCAD code being written instantly... does it have to be like that?

### ISSUE THREAD

**Issue by jglauche** [cjbissonnette/RubyScad#3]
Hi,

I played around with RubyScad today after being frustrated writing python code with Solidpython. I was certainly hoping that ruby could ease the pain of tons of brackets, but sadly RubyScad does not help with that at the current state. It very much feels like writing OpenSCAD code but be able to choose from a more powerful set of language tools.

I'd much more like an implementation where everything that you write is an object at first. Say you want a cylinder and want it translated by x=10 and rotated by 90° in X the current code would look like this:
translate(x:10){  
  rotate(x:90){
        cylinder(r:10,h:20)
    }
}

I'd love having a syntax that is more like this:
my_cylinder = cylinder(r:!0,h:20).rotate(x:90).translate(x:10)

To do that, I'd have to put all the OpenSCAD logic into classes, storing transformations like a tree. I think that I'm gonna try to implement that, your code looks very useful to actually put together the openscad output. If I proceed... would you like to put pull back the drastic changes at this point (maybe as 2.0 version) or would you prefer that I fork and rename it?

**Comment by jglauche** [comment]
Hey. I decided to build my own library based on RubyScad to be more useful to my needs: https://github.com/Joaz/CrystalScad

**Comment by cjbissonnette** [comment]
Very sorry I did not get back to you sooner, I did not get an email notification of this.  I originally wanted to make something as you suggest, but decided to make a more familiar looking syntax.  This made it easier for people coming from openscad, and also made the code much simpler.  I'm very interested to see what you come up with and wish you luck.  Another project you might want to take a look at is:
https://github.com/bluebie/oozby/
